lgc/util/llpcGfxRegHelper.cpp

```diff
@@ -0,0 +1,308 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+***********************************************************************************************************************
+* @file  llpcGfxRegHelper.cpp
+* @brief LLPC source file: Implementation of LLPC utility class GfxRegHelper
+***********************************************************************************************************************
+*/
+#include "llpcGfxRegHelper.h"
+#include "llvm/IR/IntrinsicsAMDGPU.h"
+
+using namespace lgc;
+using namespace llvm;
+
+// =====================================================================================================================
+GfxRegHelper::GfxRegHelper(
+    Builder* pBuilder,  // [in] The builder handle
+    Value*   pRegister) // [in] The registered target vec <n x i32>
+: GfxRegHandlerBase(pBuilder, pRegister)
+{
+    m_pInt32One = ConstantInt::get(m_pBuilder->getInt32Ty(), 1);
+}
+
+// =====================================================================================================================
+// Common function for getting register
+// - Cache value when fetching
+Value* GfxRegHelper::GetRegCommon(
+    uint32_t id) // [in] The Reg ID
+{
+    // Under two condition, we need to fetch the range of bits
+    //  - The register has not being initialized
+    //  - The register is being modified
+    if ((m_bitsState[id].pValue == nullptr) ||
+        (m_bitsState[id].isModified == 1))
```

Comment:
nit: `isModified` is a bool, so you can use it directly instead of doing the comparison